### PR TITLE
add probers of liveliness / readiness for ca

### PR DESCRIPTION
**Pull request by mangchiandjjoe**
liveness / readiness probes are added to ca

### CODE REVIEW

**Comment by wattli** [comment]
Please fix lint.

File: security/pkg/pki/ca/ca.go [hunk 1]
```diff
@@ -128,14 +134,17 @@ func NewSelfSignedIstioCA(caCertTTL, certTTL, maxCertTTL time.Duration, org stri
...
 }
 
 // NewIstioCA returns a new IstioCA instance.
 func NewIstioCA(opts *IstioCAOptions) (*IstioCA, error) {
 	ca := &IstioCA{
 		certTTL:    opts.CertTTL,
 		maxCertTTL: opts.MaxCertTTL,
+
+		livenessProbe:  probe.NewProbe(),
+		readinessProbe: probe.NewProbe(),
```
Comment (jmuk) [inline security/pkg/pki/ca/ca.go]: Not sure why both liveness and readiness exist here -- if they deliver the same signal, you can skip readiness for now.

Comment (mangchiandjjoe) [reply security/pkg/pki/ca/ca.go]: Removed liveness probe

Comment (mangchiandjjoe) [reply security/pkg/pki/ca/ca.go]: Reverted liveness and removed readiness

File: security/cmd/istio_ca/main.go [hunk 2]
```diff
@@ -201,30 +211,34 @@ func createClientset() *kubernetes.Clientset {
...
-		istioCA, err := ca.NewSelfSignedIstioCA(opts.caCertTTL, opts.workloadCertTTL, opts.maxWorkloadCertTTL, opts.selfSignedCAOrg,
-			opts.istioCaStorageNamespace, core)
+		caOpts, err = ca.NewSelfSignedIstioCAOptions(opts.caCertTTL, opts.workloadCertTTL,
+			opts.maxWorkloadCertTTL, opts.selfSignedCAOrg, opts.istioCaStorageNamespace, core)
 		if err != nil {
 			fatalf("Failed to create a self-signed Istio CA (error: %v)", err)
 		}
-		return istioCA
+	} else {
+		var certChainBytes []byte
```
Comment (myidpt) [inline security/cmd/istio_ca/main.go]: To make it consistent with the code above, I think you can refactor the following code to ca.NewIstioCAOptions(). You can add a TODO here.

Comment (mangchiandjjoe) [reply security/cmd/istio_ca/main.go]: Created https://github.com/istio/istio/issues/2977 for this.

**Pull request merged**